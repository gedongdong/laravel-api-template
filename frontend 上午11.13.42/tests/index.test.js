import pkg from '../package.json';

const testDist = process.env.LIB_DIR === 'dist';

describe('antd dist files', () => {
  // https://github.com/ant-design/ant-design/issues/1638
  // https://github.com/ant-design/ant-design/issues/1968
  it('exports modules correctly', () => {
    const antd = testDist ? require('../dist/antd') : require('../components'); // eslint-disable-line global-require
    expect(
      Object.keys(antd).map(key => {
        if (antd[key].displayName) {
          return `${key}: { displayName: ${antd[key].displayName} }`;
        }
        if (antd[key].name) {
          return `${key}: { name: ${antd[key].name} }`;
        }
        return key;
      }),
    ).toMatchSnapshot();
  });

  // https://github.com/ant-design/ant-design/issues/1970
  // https://github.com/ant-design/ant-design/issues/1804
  if (testDist) {
    it('should have antd.version', () => {
      const antd = require('../dist/antd'); // eslint-disable-line global-require
      expect(antd.version).toBe(pkg.version);
    });
  }
});
