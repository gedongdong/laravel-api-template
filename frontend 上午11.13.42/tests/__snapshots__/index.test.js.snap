// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`antd dist files exports modules correctly 1`] = `
Array [
  "default",
  "install: { name: install }",
  "version",
  "Affix: { name: AAffix }",
  "Alert: { name: AAlert }",
  "Anchor: { name: AAnchor }",
  "AnchorLink: { name: AAnchorLink }",
  "AutoComplete: { name: AAutoComplete }",
  "AutoCompleteOptGroup: { displayName: AAutoCompleteOptGroup }",
  "AutoCompleteOption: { displayName: AAutoCompleteOption }",
  "Avatar: { name: AAvatar }",
  "AvatarGroup: { name: AAvatarGroup }",
  "BackTop: { name: ABackTop }",
  "Badge: { name: ABadge }",
  "BadgeRibbon: { name: ABadgeRibbon }",
  "Breadcrumb: { name: ABreadcrumb }",
  "BreadcrumbItem: { name: ABreadcrumbItem }",
  "BreadcrumbSeparator: { name: ABreadcrumbSeparator }",
  "Button: { name: AButton }",
  "ButtonGroup: { name: AButtonGroup }",
  "Calendar: { name: ACalendar }",
  "Card: { name: ACard }",
  "CardGrid: { name: ACardGrid }",
  "CardMeta: { name: ACardMeta }",
  "Carousel: { name: ACarousel }",
  "Cascader: { name: ACascader }",
  "CheckableTag: { name: ACheckableTag }",
  "Checkbox: { name: ACheckbox }",
  "CheckboxGroup: { name: ACheckboxGroup }",
  "Col: { name: ACol }",
  "Collapse: { name: ACollapse }",
  "CollapsePanel: { name: ACollapsePanel }",
  "Comment: { name: AComment }",
  "ConfigProvider: { name: AConfigProvider }",
  "DatePicker: { name: ADatePicker }",
  "Descriptions: { name: ADescriptions }",
  "DescriptionsItem: { name: ADescriptionsItem }",
  "DirectoryTree: { name: ADirectoryTree }",
  "Divider: { name: ADivider }",
  "Drawer: { name: ADrawer }",
  "Dropdown: { name: ADropdown }",
  "DropdownButton: { name: ADropdownButton }",
  "Empty: { displayName: AEmpty }",
  "Form: { name: AForm }",
  "FormItem: { name: AFormItem }",
  "FormItemRest: { name: AFormItemRest }",
  "Grid",
  "Image: { name: AImage }",
  "ImagePreviewGroup: { name: AImagePreviewGroup }",
  "Input: { name: AInput }",
  "InputGroup: { name: AInputGroup }",
  "InputNumber: { name: AInputNumber }",
  "InputPassword: { name: AInputPassword }",
  "InputSearch: { name: AInputSearch }",
  "Layout: { name: ALayout }",
  "LayoutContent: { name: ALayoutContent }",
  "LayoutFooter: { name: ALayoutFooter }",
  "LayoutHeader: { name: ALayoutHeader }",
  "LayoutSider: { name: ALayoutSider }",
  "List: { name: AList }",
  "ListItem: { name: AListItem }",
  "ListItemMeta: { displayName: AListItemMeta }",
  "LocaleProvider: { name: ALocaleProvider }",
  "Mentions: { name: AMentions }",
  "MentionsOption: { name: AMentionsOption }",
  "Menu: { name: AMenu }",
  "MenuDivider: { name: AMenuDivider }",
  "MenuItem: { name: AMenuItem }",
  "MenuItemGroup: { name: AMenuItemGroup }",
  "Modal: { name: AModal }",
  "MonthPicker: { name: AMonthPicker }",
  "PageHeader: { name: APageHeader }",
  "Pagination: { name: APagination }",
  "Popconfirm: { name: APopconfirm }",
  "Popover: { name: APopover }",
  "Progress: { name: AProgress }",
  "QuarterPicker: { name: AQuarterPicker }",
  "Radio: { name: ARadio }",
  "RadioButton: { name: ARadioButton }",
  "RadioGroup: { name: ARadioGroup }",
  "RangePicker: { name: ARangePicker }",
  "Rate: { name: ARate }",
  "Result: { name: AResult }",
  "Row: { name: ARow }",
  "Select: { name: ASelect }",
  "SelectOptGroup: { displayName: ASelectOptGroup }",
  "SelectOption: { displayName: ASelectOption }",
  "Skeleton: { name: ASkeleton }",
  "SkeletonAvatar: { name: ASkeletonAvatar }",
  "SkeletonButton: { name: ASkeletonButton }",
  "SkeletonImage: { name: ASkeletonImage }",
  "SkeletonInput: { name: ASkeletonInput }",
  "SkeletonTitle: { name: SkeletonTitle }",
  "Slider: { name: ASlider }",
  "Space: { name: ASpace }",
  "Spin: { name: ASpin }",
  "Statistic: { name: AStatistic }",
  "StatisticCountdown: { name: AStatisticCountdown }",
  "Step: { name: AStep }",
  "Steps: { name: ASteps }",
  "SubMenu: { name: ASubMenu }",
  "Switch: { name: ASwitch }",
  "TabPane: { name: ATabPane }",
  "Table: { name: ATable }",
  "TableColumn: { name: ATableColumn }",
  "TableColumnGroup: { name: ATableColumnGroup }",
  "TableSummary: { name: ATableSummary }",
  "TableSummaryCell: { name: ATableSummaryCell }",
  "TableSummaryRow: { name: ATableSummaryRow }",
  "Tabs: { name: ATabs }",
  "Tag: { name: ATag }",
  "Textarea: { name: ATextarea }",
  "TimePicker: { name: ATimePicker }",
  "TimeRangePicker: { name: ATimeRangePicker }",
  "Timeline: { name: ATimeline }",
  "TimelineItem: { name: ATimelineItem }",
  "Tooltip: { name: ATooltip }",
  "Transfer: { name: ATransfer }",
  "Tree: { name: ATree }",
  "TreeNode: { name: ATreeNode }",
  "TreeSelect: { name: ATreeSelect }",
  "TreeSelectNode: { displayName: ATreeSelectNode }",
  "Typography: { name: ATypography }",
  "TypographyLink: { displayName: ATypographyLink }",
  "TypographyParagraph: { displayName: ATypographyParagraph }",
  "TypographyText: { displayName: ATypographyText }",
  "TypographyTitle: { displayName: ATypographyTitle }",
  "Upload: { name: AUpload }",
  "UploadDragger: { name: AUploadDragger }",
  "WeekPicker: { name: AWeekPicker }",
  "message",
  "notification",
]
`;
